plans/features/main.fmf

```diff
@@ -0,0 +1,15 @@
+prepare+:
+  # Install jq and yq for cleaner tests
+  - how: install
+    package:
+      - jq
+      - python3-pip
+
+  - how: shell
+    script:
+      # Try to install into .local first, if that fails, try without --user.
+      # In the sane world, tmt would be running in a virtualenv, and we'd
+      # install yq there. Otherwise, we'd change system packages, but we'd
+      # have to run as root to do that, and who's running tmt test suite
+      # as root?
+      - pip3 install --user yq || pip3 install yq
```

Comment:
Should `yq` be also added here https://github.com/teemtee/tmt/blob/main/setup.py#L72? The reason is local development, executing the tests directly, what I find handy ....

Reply:
Hard to tell: `yq` is a thin wrapper for `jq`, so we'd need to require `jq` somehow. Can't do that in `setup.py`.

Reply:
Sure, you need to have `jq` installed on your system. That is now a requirement if we want to use `yq` in the tests. Same as you would need `gcc` if there would be some python dependency requiring compilation.

Reply:
This should be extended:

https://tmt.readthedocs.io/en/stable/contribute.html#develop

Reply:
@psss WDYT?

Reply:
Hm, what about dropping the extra dependency and using the following one-liner instead?

    python -c 'import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout, indent=4)' < file.yaml > file.json

We could also create a shared python script if the `jq` queries are to be used in multiple tests.

Reply:
> This should be extended:
> 
> https://tmt.readthedocs.io/en/stable/contribute.html#develop

Sure, I'll do that.

> Hm, what about dropping the extra dependency and using the following one-liner instead?
> 
> ```
> python -c 'import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout, indent=4)' < file.yaml > file.json
> ```
> 
> We could also create a shared python script if the `jq` queries are to be used in multiple tests.

I'm not sure I follow. The issue is the extra RPM requirement, hiding under the hood of `yq`, which is `jq`. With this one-liner, we'd be able to convert a YAML file into a JSON, but what would happen then? The powerful `jq`/`yq` query language, the actual feature I'm after here, would be still missing. And yes, `jq` - or rather `yq`, because tmt emits information in YAML format - queries are to be used in many tests, to replace various existing and future `grep` calls.

Reply:
I meant let's just require `jq` which is available as `rpm` package (so we would not have to do the extra `pip install` stuff), and during tests just convert `yaml` to `json` using the above-mentioned one-liner and happily use the query language of `jq`.

Reply:
I for one would still pick `yq` over the one-liner: it works out of the box, it matches the "native language" of tmt, and it's one less item on the tmt command line in tests. Is one more Python package a price too big? My impression was the biggest offender would be the `jq` RPM we can't put in `setup.py`, rather than a Python package we can install in local development rather easily - if we agree on a way to get `jq` in for developers, adding a single Python package should be a trivial endeavor.

Reply:
I, personally, would also opt `yq`, like it is a development dependency, not runtime, so why to avoid it?

@lukaszachy WDYT?

Reply:
Isn't it just lovely? :) That query language makes the difference, and with `yq` is just simple :)

```diff
-        rlRun "grep -Pzo '(?sm)^/test/shell/good:$.*?^ *result: pass$' $results" 0 "Check pass"
+        rlRun "yq -ery '.[] | select(.name == \"/test/shell/good\" and .result == \"pass\")' $results" 0 "Check pass"
```

Reply:
Sorry, probably misunderstood, I thought you're not happy with the extra `pip` installation hack so I was trying to suggest an alternative. I'm ok with the current approach.

Reply:
IIUIC, the original comment was meant to add `yq` as a Python requirement for developers. It's already installed for tests, thanks to this change, but @thrix proposed adding it to `setup.py` as well, to make local development installation more friendly.

Reply:
Trivial but possibly good enough change in a71e909a

Reply:
Yes, that makes sense!